app/controllers/NextUpdatesController.scala

```diff
@@ -59,7 +59,7 @@ class NextUpdatesController @Inject()(NoNextUpdatesView: NoNextUpdates,
     }
   }
 
-  def getNextUpdates(backUrl: Call, isAgent: Boolean, errorHandler: ShowInternalServerError, origin: Option[String] = None)
+  def getNextUpdates(backUrl: Call, isAgent: Boolean, errorHandler: ShowInternalServerError, origin: Option[String] = None, showReportingFrequencyContent: Boolean = false)
```

Comment:
think we don't need a boolean flag here as a parameter and can just call the feature switch inside this method? 

```
val shouldShowReportingContent: Boolean = isEnabled(ReportingFrequencyPage)
```

also I try to not set default values for params unless it's for a  good reason or massive syntax convenience. Helps with spotting issues sooner with the code. Since the value is not defaulted, if the value needs to be passed around a lot it explicitly defined at each point.

Reply:
will look into all the comments ta 😄

Reply:
wonder what you think about something like this?

https://github.com/hmrc/income-tax-view-change-frontend/compare/main...MISUV-9024b